src/cargo/ops/fix.rs

```diff
@@ -130,12 +115,9 @@ pub fn fix(ws: &Workspace<'_>, opts: &mut FixOptions<'_>) -> CargoResult<()> {
         server.configure(&mut wrapper);
     }
 
-    let rustc = opts.compile_opts.config.load_global_rustc(Some(ws))?;
-    wrapper.arg(&rustc.path);
```

Comment:
I would leave this here, but set the first arg to: `wrapper.arg(&rustc.workspace_path());`.